src/array.c

```diff
@@ -1219,6 +1221,7 @@ JL_DLLEXPORT void jl_array_ptr_copy(jl_array_t *dest, void **dest_p,
         }
     }
     memmove(dest_p, src_p, n * sizeof(void*));
+    jl_fence_release(); // to finish up, ensure contents of dest is now visible on other processors too
```

Comment:
And the missing atomicity aside, I think the fence needs to go before all stores to `dest`.

Reply:
True, I should stop using memmove also, and switch to `llvm.memmove.element.unordered.atomic` to be correct. This fence just means you can use a different (relaxed) variable as a flag to indicate the operation is done.

There is also a fence added at the start, to ensure that the pointers consume-loaded in this loop will also be valid to dereference (consume-load) on another thread.